### ISSUE TITLE
Fails without setting EXECUTOR_MAX_WORKERS

### ISSUE THREAD

**Issue by srirams** [dchevell/flask-executor#5]
Using Python 2.7 on Windows, this:

[code]
def default_workers(executor_type, major=version_info.major, minor=version_info.minor):
    if not (major == 3 and minor in (3, 4)):
        return None
    return cpu_count() * WORKERS_MULTIPLIER.get(executor_type, 1)
[/code]

returns None, and the executor task never starts.

Setting EXECUTOR_MAX_WORKERS in the flask config fixes it.

**Comment by dchevell** [comment]
Hey @srirams 

This is a little strange. The reason that function returns none is that the `futures` module that backports `concurrent.futures` to Python 2 allows passing `max_workers=None` when initialising a `concurrent.futures.ThreadPoolExecutor` (as does the built in `concurrent.futures` module from Python 3.5+), and provides internal logic for choosing a default value. The function you referenced simply provides this default logic if running on Python 3.3 or 3.4, and returns `None` for all other Python versions which use a version of the module capable of choosing a default value. 

I can't reproduce this failure in Python 2:
```
$ python --version
Python 2.7.15
$ python
Python 2.7.15 (default, Jun 17 2018, 12:46:58)
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from flask import Flask
>>> from flask_executor import Executor
>>> app = Flask(__name__)
>>> executor = Executor(app)
>>> print(executor._executor._max_workers)
40
```

Can you share the following:

* An example stack trace
* The output of `pip freeze` and of `python --version`

**Comment by dchevell** [comment]
I've pinpointed what I think your issue is. Whilst the latest version of the backported `futures` module supports passing `max_workers=None`, this improvement was introduced February 2017 in [3.1.1](https://github.com/agronholm/pythonfutures/blob/3.1.1/concurrent/futures/thread.py#L93). In the previous version, [3.0.5](https://github.com/agronholm/pythonfutures/blob/3.0.5/concurrent/futures/thread.py#L85) there was no default value.

Is it an option for you to upgrade to a more recent version of the `futures` module? I could add additional checks but would prefer not to add complexity to this particular function as I ultimately want to remove the logic altogether once Python 3.4 reaches EOL.

**Comment by srirams** [comment]
Yes, that was the problem exactly. It's not an issue to for me to upgrade or set the value explicitly, but adding a note to the readme with the version requirement might save someone else a bit of debugging in the future :)
